docs/dev/versions.md

```diff
@@ -0,0 +1,56 @@
+
+
+## What is _not_ covered
+
+
+
+*   **Experimental APIs**: To facilitate development, we exempt some API symbols clearly marked as experimental from the compatibility guarantees. In particular, the following are not covered by any compatibility guarantees:
+    *   any symbol in the `cirq.contrib` module or its submodules.
+    *   any symbol (module, function, argument, property, class, or constant) whose name contains `experimental` or `Experimental`; or
+    *   any symbol whose fully qualified name includes a module or class which is itself experimental. This includes fields and submessages of any json/protocol serialization called `experimental`.
+*   **Details of gate decompositions:** Many public gates and operations in Cirq are composite gates defined by specifying a particular decomposition in terms of other simpler gates, using `_decompose_` protocol. These decompositions may change for minor releases, with a guarantee that the old and new decompositions would result in the same unitary up to a global phase.
+*   **Floating point numerical details:** The specific floating point values computed by simulators / linalg utilities may change at any time. Users should rely only on approximate accuracy and numerical stability, not on the specific bits computed.
+*   **Bugs:** We reserve the right to make backwards incompatible behavior (though not API) changes if the current implementation is clearly broken, that is, if it contradicts the documentation or if a well-known and well-defined intended behavior is not properly implemented due to a bug.
+    *   For example: if a transformer claims to implement a well-known transformation algorithm but does not match that algorithm due to a bug, then we will fix the transformer. Our fix may break code relying on the wrong behavior of the transformer. We will note such changes in the release notes.
+*   **Circuit diagrams:** We reserve the right to change exact diagrams for gates / operations / circuits across minor releases. These changes would also be explicitly announced on cirq-announce@.
+    *   For example: We can change [SVG diagrams](https://github.com/quantumlib/Cirq/issues/5689) and [text diagrams](https://github.com/quantumlib/Cirq/issues/5688) to use a different symbol for all classically controlled operations.
+*   **Error behavior:** We may replace input validation errors with non-error behavior. For instance, we may change a class/function to compute a result instead of raising an error for a certain set of inputs, even if that error is documented. We also reserve the right to change the text of error messages.
+    *   For example: If we decide to extend functionality of `PauliSumExponential` to support non-commuting Pauli’s, the [value error](https://github.com/quantumlib/Cirq/blob/e00767a2ef1233e82e9089cf3801a77e4cc3aea3/cirq-core/cirq/ops/pauli_sum_exponential.py#L53) in the constructor would be removed.
+
```

Comment:
What do you plan to guarantee around type preservation? Should types always be conserved, or is Liskov Substitution Principle good enough?

i.e. if there's a function `f()->BaseClass` that returns `Subclass1`, then could a new version change that to return `Subclass2`, so long as that object behaves the same from the `BaseClass` interface?

Reply:
I was going to ask a similar question. If a function used to return a `np.int64` can we change it to return `int`? This is like 99% backwards compatible but not strictly so

Reply:
> Should types always be conserved, or is Liskov Substitution Principle good enough?
Satisfying LSP is good enough and we should guarantee type preservation. This is because ideally we would like the policies to be flexible enough to add new features while ensuring that existing code works -- and satisfying LSP guarantees this.

Of course, if the function definition has `f() -> Subclass1` and you change it to return `f() -> Subclass2`; then it's not allowed and it doesn't satisfy LSP. 

 > If a function used to return a np.int64 can we change it to return int

This would not satisfy LSP and should be considered a breaking the change. The inverse however is fine (i.e. return `np.int64` where we currently return `np.int`) since with return types; substituting a more general type to replace a more restrictive type is fine (everything that works with `np.int` should continue to work with `np.int64) but the inverse is not true. 


Added a bullet point to highlight the specific case of type preservation. Thanks for highlighting!